resolver.go

```diff
@@ -388,7 +391,7 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 
 			// forwardQueryStart stores required context to mux multiple client queries over
 			// one connection; and limits the number of outstanding concurrent queries.
-			if r.forwardQueryStart(w, query) == false {
+			if r.forwardQueryStart(w, query, queryID) == false {
```

Comment:
The passed `query` contains the ID. And queryID seems to initialized to `query.ID` above so I don't understand the need to pass it separately. What is it fixing?